### DISCUSSION TITLE
Question about Overlapping grids

### DISCUSSION THREAD

**Discussion by lacombef** [mfem/mfem#3371] [General]
Hi, 

I am currently looking to use MFEM as a basis for various multiphysics problems. I know there is a possibility to use subdomains, but I am also considering to use the overlapping grids framework to couple different physics on different grid entities; I find it very clever to use different grids depending on the physics of the problem, particularly if you use AMR as the resulting grids will be truly adapted to each problem.

I was just wondering about the implementation of the overlapping grid method. Is there a theoretical description of the method somewhere, for example a paper to which I could refer ? My main concern is related to the conservation of mass and momentum. For instance, in fluid mechanics, if the flow field is simply projected onto another grid, the resulting flow might not conserve mass and/or momentum.

Thank you !

**Comment by tzanio** [comment]
ping: @jandrej, @psocratis

**Reply by jandrej** [reply]
Oh I'd ping @kmittal2 here :)

**Comment by kmittal2** [comment]
Hi @lacombef,

The theoretical description of overlapping grid method that you are looking will depend on the PDE that you are interested in solving. For example, for the incompressible Navier-Stokes equations (INSE), the approach for time-stepping is describe [here](https://www.sciencedirect.com/science/article/pii/S0021999115008050) and [here](https://www.sciencedirect.com/science/article/pii/S0021999121002308), and the approach for ensuring Mass conservation is described [here](https://www.sciencedirect.com/science/article/pii/S0045793018308326).
There are also papers on other systems such as fluid structure interactions and Maxwell equations by Henshaw et al. So I would suggest taking a look at the literature based on the PDE of interest.

Right now we have the general mechanism for solution interpolation and communication between overlapping grids in MFEM, but have not implemented any constraints (e.g. mass conservation for INSE). However, I will be happy to answer your questions if you are interested in trying it.

Ketan

**Reply by lacombef** [reply]
Hi Ketan,

thank you for your answer. I will look into that. So, if I understand correctly, at the moment, in the overlapping grid method, the solution is interpolated from one grid to the other, but there are no constraint for mass or momentum conservation. Although, it is possible to implement it. It's good to know.

Thanks,

Francis
